srml/democracy/src/lib.rs

```diff
@@ -278,29 +314,58 @@ impl<T: Trait> Module<T> {
 						<balances::Module<T>>::unreserve(d, deposit);
 					}
 					Self::deposit_event(RawEvent::Tabled(prop_index, deposit, depositors));
-					Self::inject_referendum(now + Self::voting_period(), proposal, VoteThreshold::SuperMajorityApprove)?;
+					Self::inject_referendum(now + Self::voting_period(), proposal, VoteThreshold::SuperMajorityApprove, Self::public_delay())?;
 				}
 			}
 		}
 
 		// tally up votes for any expiring referenda.
-		for (index, _, proposal, vote_threshold) in Self::maturing_referendums_at(now) {
+		for (index, _, proposal, threshold, delay) in Self::maturing_referendums_at(now) {
 			let (approve, against) = Self::tally(index);
 			let total_issuance = <balances::Module<T>>::total_issuance();
+			let approved = threshold.approved(approve, against, total_issuance);
+
+			Self::voters_for(index).into_iter()
+				.filter(|a| (Self::vote_of((index, a.clone())).unwrap_or(false)/*defensive only: all items come from `voters`; for an item to be in `voters` there must be a vote registered; qed*/ == approved))
+				.for_each(|a| <Bondage<T>>::mutate(a, |b| if *b < now + delay { *b = now + delay }));
```

Comment:
should it be `*b <= now` so it'll also apply the delay to accounts who's funds became liquid at the current block (`now`) as well?